composer.json

```diff
@@ -25,7 +25,8 @@
 		"php": ">=8.1",
 		"composer/installers": "^2|^1.0.1",
 		"opis/json-schema": "^2.3.0",
-		"ruflin/elastica":  "^7.1.0"
+		"ruflin/elastica":  "^7.1.0",
+		"psr/log": "^1.1.4"
```

Comment:
The elastica import was pulling in psr/log 3.x. Meanwhile, MW requires that to be at 1.1.4, due to method signature imcompatibilities. This was causing the previous CI failures.

Reply:
So the reason you need a new desk is that MediaWiki devs can't define their dependencies?

Bit of a weird approach in MediaWiki. Either you get inappropriate pinning like `5.21.6.47.1.9.52.3`, or the constraint is missing.

Reply:
MW actually pins it as "1.1.4", but we're getting composer dependencies in both MW and separately in the extension. MW's `vendor` has the right version. I think something weird was going on with PHPStan using (some of?) the extension's vendors. So it ends up analysing MW code using a wrong version of psr/log, because ruflin/elastica allows 1|2|3.

Reply:
We might be installing PHPStan in the extension itself, ignoring MW dependencies. Such an approach is often needed to avoid conflicts with those MediaWiki pins or to be able to use PHPUnit that is not from 2007.